xbout/calc/tests/test_turbulence.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+from xarray import DataArray
+import pytest
+import numpy.testing as npt
+
+from xbout.calc.turbulence import rms
+
+
+class TestRootMeanSquare:
+    def test_no_dim(self):
+        dat = np.array([5, 7, 3.2, -1, -4.4])
+        orig = DataArray(dat, dims=['x'])
+
+        # Need to supply dimension
+        with pytest.raises(ValueError):
+            rms(orig, dim=None)
+
+    def test_1d(self):
+        dat = np.array([5, 7, 3.2, -1, -4.4])
+        orig = DataArray(dat, dims=['x'])
+
+        sum_squares = np.sum(dat**2)
+        mean_squares = sum_squares/dat.size
+        rootmeansquare = np.sqrt(mean_squares)
+
+        expected = rootmeansquare
+        actual = rms(orig, dim='x').values
+        npt.assert_equal(actual, expected)
```

Comment:
Should we use `npt.assert_almost_equal(actual, expected, decimal=15)` instead?

Reply:
Not sure - I mean the tests pass, is there a reason to expect that wouldn't always be the case? Is this an operation which would be affected by the precision used? Both the test and the function are still using numpy

Reply:
In this particular case, I don't know why the test would ever fail. I think the usual practice for this kind of test is to include a small tolerance to allow for rounding errors. I think the reason might be that if for some reason (eg. optimisation) we, for example, change the order of some operations, we might get a result that is mathematically correct, but not identical to the last bit. So it's 'good practice' to include a small tolerance.

Reply:
I think `assert_equal` is fine for now. We can fix it easily if it ever fails with rounding-error changes for anyone.